Apps/Database/src/Delegates/DBCommunicationDelegate.cs

```diff
@@ -146,5 +146,34 @@ public List<Communication> GetEmailCommunicationsToSend()
             this.logger.LogDebug($"Finished getting list of New & Processing Email Communications from DB. {JsonSerializer.Serialize(retVal)}");
             return retVal;
         }
+
+        /// <inheritdoc />
+        public DBResult<Communication> Delete(Communication communication, bool commit = true)
+        {
+            this.logger.LogTrace($"Deleting Communication from DB...");
+            DBResult<Communication> result = new DBResult<Communication>()
+            {
+                Payload = communication,
+                Status = DBStatusCode.Deferred,
+            };
+            this.dbContext.Communication.Remove(communication);
```

Comment:
should it check that the communication has not been processed yet?

Reply:
The service should likely check that logic instead of having the delegate.